### DISCUSSION TITLE
RemoveContainedItemsOnDeconstruct not working for second ItemContainer

### DISCUSSION THREAD

**Discussion by zhu-rengong** [FakeFishGames/Barotrauma#14424] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

ItemContainer's `RemoveContainedItemsOnDeconstruct` will prevent the items in its Inventory from being transferred to the Deconstructor's output slot and remove them after deconstruction. But it only works for the first ItemContainer, because `Entity.Spawner.AddItemToRemoveQueue` does not take secondary Containers into account.

**Relevant Code**

https://github.com/FakeFishGames/Barotrauma/blob/0aee53976202c7ea5c0bb3f26669a97c8dc2400c/Barotrauma/BarotraumaShared/SharedSource/Items/Components/Machines/Deconstructor.cs#L338-L357

https://github.com/FakeFishGames/Barotrauma/blob/0aee53976202c7ea5c0bb3f26669a97c8dc2400c/Barotrauma/BarotraumaShared/SharedSource/Networking/EntitySpawner.cs#L367-L382

### Reproduction steps

1. Set `RemoveContainedItemsOnDeconstruct` of secondary item container for  `clownexosuit` to true
2. Deconstruct the `clownexosuit`
3. Banana peels in the `clownexosuit` will fall to the ground, but it expects to be removed.

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Single player

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Fixed in https://github.com/FakeFishGames/Barotrauma-development/commit/08f8615bf0e9dc85e5d723cdff42b50611cd96f5
